src/Components/Web.JS/src/Services/WebRootComponentManager.ts

```diff
@@ -164,6 +165,21 @@ export class WebRootComponentManager implements DescriptorHandler, NavigationEnh
     }
   }
 
+    private updateApplicationState() {
+      if (isRendererAttached(WebRendererId.Server)) {
+        const serverAppState = discoverServerPersistedState(document);
+        if (serverAppState) {
+            updateApplicationState(WebRendererId.Server, serverAppState);
+        }
+      }   
+      if (isRendererAttached(WebRendererId.WebAssembly)) {
+        const webAssemblyAppState = discoverWebAssemblyPersistedState(document);
+        if (webAssemblyAppState) {
+            updateApplicationState(WebRendererId.WebAssembly, webAssemblyAppState);
+        }
+      }  
+    }
```

Comment:
I wonder if there are things that we can do to avoid scanning the entire DOM each time the document receives an SSR update.

With component comments, we scan the incoming DOM before it gets merged with the document. That allows us to only scan the subset of the document that gets changed. Could we consider a similar approach here?

Reply:
Agree. Working on it